rust/lance/src/datafusion/logical_expr.rs

```diff
@@ -39,6 +46,49 @@ fn resolve_value(expr: &Expr, data_type: &DataType) -> Result<Expr> {
     }
 }
 
+/// Given a Expr::Column or Expr::GetIndexedField, get the data type of referenced
+/// field in the schema.
+///
+/// If the column is not found in the schema, return None. If the expression is
+/// not a field reference, also returns None.
+pub fn resolve_column_type(expr: &Expr, schema: &Schema) -> Option<DataType> {
+    let mut field_path = VecDeque::new();
```

Comment:
It doesn't really matter but you can probably just use a `Vec` here right?

Reply:
Yes, I guess I can just reverse the vector.